backend/pipelines/dmi_pipeline/bronze/extract.py

```diff
@@ -0,0 +1,111 @@
+"""
+DMI Climate Data Extraction Layer
+Handles API communication and data retrieval from the Danish Meteorological Institute.
+Saves raw data without any transformations.
+"""
+
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Any
+import aiohttp
+from fastapi import HTTPException
+import geopandas as gpd
+from shapely.geometry import shape
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+)
+logger = logging.getLogger(__name__)
+
+class DMIConfig:
+    """Configuration for DMI API access"""
+    def __init__(self):
+        self.api_key = os.getenv('DMI_GOV_CLOUD_API_KEY')
+        if not self.api_key:
+            raise ValueError("DMI_GOV_CLOUD_API_KEY environment variable is required")
+
+        self.base_url = "https://dmigw.govcloud.dk/v2/climateData"
+        self.max_retries = int(os.getenv('MAX_RETRIES', 3))
+        self.retry_delay = int(os.getenv('RETRY_DELAY', 5))
+        self.SOURCE_CRS = "EPSG:25832"  # DMI's native CRS
+
+class DMIApiClient:
+    """Client for interacting with DMI's climate data API"""
+    def __init__(self, config: DMIConfig):
+        self.config = config
+        self.headers = {
+            "Accept": "application/geo+json",
+            "X-Gravitee-Api-Key": self.config.api_key
+        }
+
+    async def _make_request(self, endpoint: str, params: Dict[str, Any] = None) -> Dict:
+        """Make an authenticated request to the DMI API with error handling"""
+        if params is None:
+            params = {}
+
+        url = f"{self.config.base_url}/{endpoint}"
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, headers=self.headers, params=params) as response:
+                    if response.status == 429:
+                        logger.warning("Rate limit exceeded")
+                        raise HTTPException(status_code=429, detail="Rate limit exceeded")
+
+                    text = await response.text()
+                    if not text:
+                        raise HTTPException(status_code=500, detail="Empty response from DMI API")
+
+                    return await response.json()
+            except Exception as e:
+                logger.error(f"Error making request to DMI API: {str(e)}")
+                raise
+
+    async def fetch_grid_data(self, parameter_id: str, start_time: datetime, end_time: datetime) -> gpd.GeoDataFrame:
```

Comment:
can't you just store it as json instead of converting to gdf and shapely? (in bronze)